src/elfpy/markets.py

```diff
@@ -115,42 +124,56 @@ def check_fees(
                 + state_string
             )
 
-    def update_market(self, market_deltas):
+    def update_market(self, market_deltas, trade_type):
         """
         Increments member variables to reflect current market conditions
         """
         for key, value in market_deltas.items():
-            assert np.isfinite(value), f"markets.update_market: ERROR: market delta key {key} is not finite."
-        self.share_reserves += market_deltas["d_base_asset"]
+            if key == "d_liquidity_pool_history":
+                assert isinstance(value, list), f"markets.updatE_market: Error:"\
+                + f" d_liquidity_pool_history has value={value} should be a list"
+            else:
+                assert np.isfinite(value), f"markets.update_market: ERROR: market delta key {key} is not finite."
+        self.share_reserves += market_deltas["d_base_asset"]/self.share_price
         self.bond_reserves += market_deltas["d_token_asset"]
-        self.cum_base_asset_slippage += market_deltas["d_base_asset_slippage"]
-        self.cum_token_asset_slippage += market_deltas["d_token_asset_slippage"]
-        self.cum_base_asset_fees += market_deltas["d_base_asset_fee"]
-        self.cum_token_asset_fees += market_deltas["d_token_asset_fee"]
-        self.base_asset_orders += market_deltas["d_base_asset_orders"]
-        self.token_asset_orders += market_deltas["d_token_asset_orders"]
-        self.base_asset_volume += market_deltas["d_base_asset_volume"]
-        self.token_asset_volume += market_deltas["d_token_asset_volume"]
-
-    def swap(self, user_action):
+        self.share_buffer += market_deltas["d_share_buffer"] if "d_share_buffer" in market_deltas else 0
+        self.bond_buffer += market_deltas["d_bond_buffer"] if "d_bond_buffer" in market_deltas else 0
+        if trade_type in ["add_liquidity", "remove_liquidity"]:
```

Comment:
If you're already checking that `"d_liquidity_pool"` is in `market_deltas` then why do you also need to check the value of `trade_type`? Couldn't we keep the function signature the same and check for `"d_liquidity_pool"` and then set += to either `market_deltas["d_liquidity_pool"]` or `0`?

Also could we set the other deltas to 0 in the case of adding liquidity? so it always looks like a "swap" to simplify this logic?